rasterio/rio/info.py

```diff
@@ -289,7 +289,7 @@ def info(ctx, input, aspect, indent, namespace, meta_member, verbose, bidx,
 @click.option(
     '-m',
     '--mode',
-    type=click.Choice(['r', 'r+']),
+    type=click.Choice(['r', 'r+', 'r-']),
```

Comment:
r- mode (no image reads, metadata only) is handy/thrifty for rio-insp'ing a dataset on S3.